src/domain/lotto-rank.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * @typedef {Object} LottoRank
+ * @property {number} match
+ * @property {boolean} hasBonus
+ * @property {number} prize
+ */
+export const LottoRank = Object.freeze({
```

Comment:
freeze를 해주셨군요 👍🏻
그런데, 어차피 외부에서 사용될 상수라면 `LOTTO_RANK`로 모두 상수화해주는 것은 어떨까요?

```suggestion
export const LOTTO_RANK = Object.freeze({
```

Reply:
@sannimdev 
코틀린에서 사용하는 enum을 자바스크립트에서는 어떻게 사용하는지 검색을 해보고 작성을 했는데요.
Object이니까 객체라 생각하고 네이밍을 하였네요. 😅 
관련해서 조금 햇갈리는데 혹시 어느 부분을 좀 더 살펴보면 이해할 수 있을까요!?
f5f635511542131cd3c063d5b0387027d8f119f0

Reply:
[대문자 사용](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Statements/const)에 관한 예시를 살펴보시면 좋을 거 같아요! 상수로 초기에 할당되는 녀석들은 재할당이 불가능하기 때문에 자바스크립트에서는 보통 모두 대문자로 사용하고 있고 단어 사이마다 언더바(\_)로 구분합니다.

```
// 주의: 상수 선언에는 대소문자 모두 사용할 수 있지만,
// 일반적인 관습은 모두 대문자를 사용하는 것입니다.

// MY_FAV를 상수로 정의하고 그 값을 7로 함
const MY_FAV = 7;

// 에러가 발생함
MY_FAV = 20;

// 7 출력
console.log("my favorite number is: " + MY_FAV);

// 상수를 재선언하려는 시도는 오류 발생 - Uncaught SyntaxError: Identifier 'MY_FAV' has already been declared
const MY_FAV = 20;

// MY_FAV라는 이름은 위에서 상수로 예약되어 있어서 역시 실패함.
var MY_FAV = 20;

// 역시 오류가 발생함
let MY_FAV = 20;

// 블록 범위의 특성을 아는게 중요
if (MY_FAV === 7) {
  // 블록 범위로 지정된 MY_FAV 라는 변수를 만드므로 괜찮습니다
  // (let으로 const 변수가 아닌 블록 범위를 선언하는 것과 똑같이 동작합니다)
  let MY_FAV = 20;

  // MY_FAV는 이제 20입니다
  console.log("my favorite number is " + MY_FAV);

  // 이 선언은 전역으로 호이스트되고 에러가 발생합니다.
  var MY_FAV = 20;
}

// MY_FAV는 여전히 7
console.log("my favorite number is " + MY_FAV);

// const 선언시에 초기값을 생략해서 오류 발생
const FOO;

// const는 오브젝트에도 잘 동작합니다
const MY_OBJECT = { key: "value" };

// 오브젝트를 덮어쓰면 오류가 발생합니다
MY_OBJECT = { OTHER_KEY: "value" };

// 하지만 오브젝트의 키는 보호되지 않습니다.
// 그러므로 아래 문장은 문제없이 실행됩니다
MY_OBJECT.key = "otherValue"; // 오브젝트를 변경할 수 없게 하려면 Object.freeze() 를 사용해야 합니다

// 배열에도 똑같이 적용됩니다
const MY_ARRAY = [];
// 배열에 아이템을 삽입하는 건 가능합니다
MY_ARRAY.push("A"); // ["A"]
// 하지만 변수에 새로운 배열을 배정하면 에러가 발생합니다
MY_ARRAY = ["B"];
```

Reply:
@sannimdev 
객체냐 아니냐에 상관없이 const로 정의한다면 상수이기 때문에 대문자라는 의미군요!
답변 감사합니다. 🙇